packages/unchained-client/src/parser/thorchain.ts

```diff
@@ -119,69 +122,77 @@ const getLiquidityType = (pool: string): LiquidityType => (pool.includes('/') ?
 const getSwapType = (memo: string): SwapType =>
   streamingSwapRegex.test(memo) ? 'Streaming' : 'Standard'
 
-export class Parser {
+export class Parser<T extends 'thorchain' | 'mayachain' = 'thorchain'> {
```

Comment:
could really use a base class, but I didn't have a good name for it so just went with this for now